Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp

```diff
@@ -0,0 +1,479 @@
+              if (hit_indices[layer] >= 0) {
+                protoTrack.push_back(hitIndicesAll[layer][hit_indices[layer]]);
+              }
+            }
+            protoTracks.push_back(protoTrack);
+          }
+        }
+  }
+  ACTS_DEBUG("Created " << protoTracks.size() << " track seeds");
+
+  ctx.eventStore.add(m_cfg.outputProtoTracks, ProtoTrackContainer{protoTracks});
+  // clear the vector 
+  houghMeasurementStructs.clear();
+  return ActsExamples::ProcessCode::SUCCESS;
+}
+
+ActsExamples::HoughHist ActsExamples::HoughTransformSeeder::createLayerHoughHist(
+    unsigned layer, int subregion) const {
+
+  ActsExamples::HoughHist houghHist(m_cfg.houghHistSize_y, m_cfg.houghHistSize_x);
+
+  for (unsigned index = 0; index < houghMeasurementStructs.size(); index++) {
+     HoughMeasurementStruct *meas = houghMeasurementStructs[index].get();
+    if (meas->layer != layer)
+      continue;
+    if (!(m_cfg.sliceTester(meas->z, meas->layer, subregion)).value())
+      continue;
+
+    // This scans over y (pT) because that is more efficient in memory
+    for (unsigned y_ = 0; y_ < m_cfg.houghHistSize_y; y_++) {
+      unsigned y_bin_min = y_;
+      unsigned y_bin_max = (y_ + 1);
+
+      // Find the min/max x bins
+      auto xBins =
+          yToXBins(y_bin_min, y_bin_max, meas->radius, meas->phi, meas->layer);
+      // Update the houghHist
+      for (unsigned y = y_bin_min; y < y_bin_max; y++)
+        for (unsigned x = xBins.first; x < xBins.second; x++) {
+          houghHist(y, x).first++;
+          houghHist(y, x).second.insert(index);
+        }
+    }
+  }
+
+  return houghHist;
+}
+
+ActsExamples::HoughHist ActsExamples::HoughTransformSeeder::createHoughHist(
+    int subregion) const {
+  ActsExamples::HoughHist houghHist(m_cfg.houghHistSize_y, m_cfg.houghHistSize_x);
+
+  for (unsigned i = 0; i < m_cfg.nLayers; i++) {
+    HoughHist layerHoughHist = createLayerHoughHist(i, subregion);
+    for (unsigned x = 0; x < m_cfg.houghHistSize_x; ++x)
+      for (unsigned y = 0; y < m_cfg.houghHistSize_y; ++y)
+        if (layerHoughHist(y, x).first > 0) {
+          houghHist(y, x).first++;
+          houghHist(y, x).second.insert(layerHoughHist(y, x).second.begin(),
+                                    layerHoughHist(y, x).second.end());
+        }
+  }
+
+  return houghHist;
+}
+
+bool ActsExamples::HoughTransformSeeder::passThreshold(HoughHist const& houghHist,
+                                                       unsigned x,
+                                                       unsigned y) const {
+  // Pass window threshold
+  unsigned width = m_cfg.threshold.size() / 2;
+  if (x < width || (houghHist.size(1) - x) < width)
+    return false;
+  for (unsigned i = 0; i < m_cfg.threshold.size(); i++) {
+    if (houghHist(y, x - width + i).first < m_cfg.threshold[i])
+      return false;
+  }
+
+  // Pass local-maximum check, if used
+  if (m_cfg.localMaxWindowSize)
```

Comment:
In our naming conventions we have decided to always use brackets, even if not necessary, so instead of writing 
```c++
for (cond1)
  for(cond2)
    do_something();
```

We would advertise to do 
```c++
for (cond1){
  for (cond2){
   doItAgain();  
}
}
```